app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -188,82 +188,57 @@ def update
-            first_delegated: role.metadata.first_delegated,
-            last_delegated: role.metadata.last_delegated,
-            total_delegated: role.metadata.total_delegated,
-          )
-          role = UserRole.create!(
-            user_id: role.user.id,
-            group_id: role.group.id,
-            start_date: Date.today,
-            metadata: metadata,
-          )
-        end
-      elsif params.key?(:groupId)
+      if params.key?(:groupId)
         group_id = params.require(:groupId)
         changes << UserRole::UserRoleChange.new(
           changed_parameter: 'Delegate Region',
           previous_value: UserGroup.find(role.group.id).name,
           new_value: UserGroup.find(group_id).name,
         )
+        new_role.group_id = group_id
+      end
 
-        return head :unauthorized unless current_user.has_permission?(:can_edit_groups, group_id)
-
-        ActiveRecord::Base.transaction do
-          role.update!(end_date: Date.today)
-          metadata = RolesMetadataDelegateRegions.create!(
-            status: role.metadata.status,
-            location: role.metadata.location,
-            first_delegated: role.metadata.first_delegated,
-            last_delegated: role.metadata.last_delegated,
-            total_delegated: role.metadata.total_delegated,
-          )
-          role = UserRole.create!(
-            user_id: role.user.id,
-            group_id: group_id,
-            start_date: Date.today,
-            metadata: metadata,
-          )
-        end
-      elsif params.key?(:location)
+      if params.key?(:location)
         location = params.require(:location)
         changes << UserRole::UserRoleChange.new(
           changed_parameter: 'Location',
           previous_value: role.metadata.location,
           new_value: location,
         )
+        new_role_metadata.location = location
+      end
 
-        ActiveRecord::Base.transaction do
-          role.update!(end_date: Date.today)
-          metadata = RolesMetadataDelegateRegions.create!(
-            status: role.metadata.status,
-            location: location,
-            first_delegated: role.metadata.first_delegated,
-            last_delegated: role.metadata.last_delegated,
-            total_delegated: role.metadata.total_delegated,
-          )
-          role = UserRole.create!(
-            user_id: role.user.id,
-            group_id: role.group.id,
-            start_date: Date.today,
-            metadata: metadata,
-          )
-        end
-      else
-        return render status: :unprocessable_entity, json: { error: "Invalid parameter to be changed" }
+      return render status: :unprocessable_entity, json: { error: "No valid parameters to be changed" } if changes.empty?
+      return head :unauthorized unless current_user.has_permission?(:can_edit_groups, new_role.group_id)
+
+      ActiveRecord::Base.transaction do
+        role.update!(end_date: Date.today)
+        new_role_metadata.save!
```

Comment:
This `save!` is redundant. Rails will automatically save all nested models during the `new_role.save!` call below.

Reply:
Done, thank you